test/factory/common_factory_tests.dart

```diff
@@ -185,18 +216,79 @@ void domEventHandlerWrappingTests(ReactComponentFactoryProxy factory) {
             'renders a single DOM node and passes props.onClick to it');
   });
 
-  test('wraps the handler with a function that converts the synthetic event', () {
-    var actualEvent;
+  group('wraps event handlers properly,', () {
+    const dart = EventTestCase.dart('onMouseUp', 'set on component');
+    const dartCloned = EventTestCase.dart('onMouseLeave', 'cloned onto component');
+    const jsCloned = EventTestCase.js('onClick', 'cloned onto component ');
+    const eventCases = {dart, jsCloned, dartCloned};
 
-    final nodeWithClickHandler = renderAndGetRootNode(factory({
-      'onClick': (event) {
-        actualEvent = event;
+    Element node;
+    Map<EventTestCase, dynamic> events;
+    Map propsFromDartRender;
+
+    setUpAll(() {
+      expect(eventCases.map((h) => h.eventPropKey).toSet(), hasLength(eventCases.length),
+          reason: 'test setup: each helper should have a unique event key');
+
+      node = null;
+      events = {};
+      propsFromDartRender = null;
+
+      var element = factory({
+        'onDartRender': (p) {
+          propsFromDartRender = p;
+        },
+        dart.eventPropKey: (event) => events[dart] = event,
+      });
+
+      element = React.cloneElement(
+        element,
+        jsifyAndAllowInterop({
+          jsCloned.eventPropKey: (event) => events[jsCloned] = event,
+        }),
```

Comment:
This would be if a JS component clones a handler onto a ReactElement created by a Dart component.

For example, passing a Dart HOC into a MUI component that clones the HOC.

I'll add a comment!

Reply:
Addressed in 70ebf83